src/octree/octree_test.rs

```diff
@@ -138,8 +145,8 @@ mod tests {
             .try_for_each_batch(callback_func)
             .expect_err("Test error");
 
-        assert_eq!(3, callback_count);
-        assert_eq!(3 * batch_size, delivered_points);
+        assert!(callback_count >= 3);
```

Comment:
I have an idea how we can avoid sending half-finished batches (but that can be a future PR): keeping a `mut Option<PointData>` in the receiver thread that can store incomplete batches. When you get an incomplete batch, append it to that PointData, and if the size of that PointData becomes >= batch_size, split off a "full" batch (i.e. of size batch_size), call the callback on that, and set the incomplete batch to the rest of the PointData.

Reply:
sounds good!